### DISCUSSION TITLE
Fire Rework (Smoke, Difficulty, Equipment)

### DISCUSSION THREAD

**Discussion by Cogs16** [FakeFishGames/Barotrauma#5666] [Ideas and suggestions]
There could be improvements made to fire as an emergency. Right now, fire is something that can be dealt with quickly by one person with a fire extinguisher with little to no preparation. I think a fire should be much more noticeable and should require more equipment to be dealt with.

-Fire should produce smoke which spreads throughout the submarine quickly and obscures vision dramatically
-Smoke should produce inhalation injuries, which are cured by wearing an oxygen mask or dive suit.
-While this is somewhat unrealistic, for the sake of fun, fires should break doors down and not hulls, additionally smoke should spread through the airvents system unless they are on or off.
-Flame particles should not be seen through walls.
-Fires consume oxygen in rooms linked by the ventilation system.

Equipment:
-Fire extinguisher should be nerfed and should be only good for extinguishing fires that are 15 seconds young.
-Fire fighting uniform/helemet. A firefighting suit which protects the user from fires and also enables better vision in  smoke.
-New heavy uty fire kit. In order to counteract the nerf of the old fire extinguisher, a new backpack-mounted version should be added which is essentially as strong as the previous fire extinguisher.

With these changes, fires should become more of a threat, which requires people to search the submarine for emergency equipment, locate the fire, and coordinate in order to successfully extinguish it, as opposed to one person running in with a fire extinguisher.

**Comment by thegrb93** [comment]
Yeah, fire was a lot more dangerous before it got nerfed. I think it got nerfed too hard.

**Comment by fracter2** [comment]
Sounds like it would work well as a mod!

**Comment by Hinny** [comment]
Firefight suit is very appealing to me. I hope that is implemented.

**Comment by Saevon** [comment]
This would work well with something like: #6877 or anything focusing on reworking oxygen to actually be a system to monitor and maintain…

1. fire burning oxygen means you need oxygen masks to fight a larger one!  
2. Smoke could be drawn in thru the vents (reversed), helping clear a room (but remove oxygen)
3. fire could be suppressed by turning off oxygen to a vent in the room, or better yet setting it to reverse?   (or would that spread fire thru vents)
4. suppressed fire could spontaneously combust if a door is opened giving it new oxygen (not major damage, but knockback and spreads fire outwards.

**Reply by Arctodites** [reply]
It would be exceedingly useful if the oxygen generators could have _'toggle_state'_ and _'set_state'_ input pins on their connection panels, because right now it is cumbersome (or, in the case of locked default wires, nigh impossible) to build circuit breakers for them, and building anything more basic than an on/off switch for them quickly gets out of hand in terms of the number of circuit components required.

For instance, I have a backup O2 generator on my custom sub that's linked to its own, separate vent subsystem that only supplies certain mission-critical rooms, and the ship uses a system of overdoor horizontal ducts to further regulate airflow between compartments.

I like the idea of being able to turn off the O2 generator to suppress a fire, but I worry that being able to turn vents off individually could make things too easy: it's fun to have to weigh-up the decision of whether to turn off the entire air supply to the whole ship versus let a fire potentially spread out of control.

Smoke that reduces visibility and causes inhalation injuries would be a great addition to the game.  One bonus to enabling smoke to spread via air vents, is that this same mechanism would be perfect for spreading other things via vents ... I could imagine traitor missions in which aerosolized drugs or acids could be placed in the oxygen tank refill slots of the O2 generator and spread to other parts of the ship via the air vents.  Maybe clowns could even have a faction-specific nitrous oxide ('laughing gas') canister for this purpose?
